lib/services/workflow_runs.py

```diff
@@ -159,27 +158,46 @@ async def get_project_workflow_runs(project_id: str) -> List[WorkflowRun]:
     return runs
 
 
+async def get_full_workflow_state(
+    thread_id: str, type: WorkflowRunType
+) -> WorkflowState:
+    """Get the full workflow state (not summary) for operations that need full data."""
+    async with get_checkpointer() as checkpointer:
+        graph = create_graph(type)
+        app = graph.compile(checkpointer=checkpointer)
+        state_snapshot = await app.aget_state(
+            {"configurable": {"thread_id": thread_id}}
+        )
+
+    if not state_snapshot or not state_snapshot.values:
+        raise HTTPException(status_code=404, detail="Workflow state not found")
+
+    state_type = get_state_type(type, summary=False)
+    full_state = _convert_state_snapshot(state_snapshot, state_type)
+    if full_state is None:
+        raise HTTPException(status_code=404, detail="Could not parse workflow state")
+
+    # Compute ranked issues for claim substantiation if not already present
+    if type == WorkflowRunType.CLAIM_SUBSTANTIATION and not full_state.ranked_issues:
+        full_state.ranked_issues = rank_issues(full_state).get("ranked_issues", [])
+
+    return full_state
```

Comment:
instead of a new method, maybe it's easier to add a param `summary: bool = True` to get_workflow_run_state_by_thread_id to use the summary state or not, similar to what `get_state_type` already does